modules/extension/wmts/src/main/java/org/geotools/ows/wmts/WMTSHelper.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2021, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.ows.wmts;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.Map;
+
+/**
+ * Some useful functions that was used within gt-wmts.
+ *
+ * <p>Might have their substitution elsewhere
+ *
+ * @author Roar Brænden
+ */
+public class WMTSHelper {
+
+    public static String replaceToken(String baseUrl, String dimName, String dimValue) {
```

Comment:
Javadoc on public members. I was going to suggest to make the class package private, but it's used in two other packages, so not an avenue